lib/core/src/Cardano/Wallet/Api/Server.hs

```diff
@@ -798,6 +828,182 @@ mkShelleyWallet ctx wid cp meta pending progress = do
+    -> ((SomeMnemonic, Maybe SomeMnemonic) -> Passphrase "encryption" -> k 'RootK XPrv)
+    -> (XPub -> k 'AccountK XPub)
+    -> ApiSharedWalletPostData
+    -> Handler ApiSharedWallet
+postSharedWallet ctx generateKey liftKey postData =
+    case postData of
+        ApiSharedWalletPostData (Left body) ->
+            postSharedWalletFromRootXPrv ctx generateKey body
+        ApiSharedWalletPostData (Right body) ->
+            postSharedWalletFromAccountXPub ctx liftKey body
+
+postSharedWalletFromRootXPrv
+    :: forall ctx s k n.
+        ( s ~ SharedState n k
+        , ctx ~ ApiLayer s k
+        , SoftDerivation k
+        , MkKeyFingerprint k (Proxy n, k 'AddressK XPub)
+        , MkKeyFingerprint k Address
+        , WalletKey k
+        , HasDBFactory s k ctx
+        , HasWorkerRegistry s k ctx
+        , Typeable s
+        , Typeable n
+        )
+    => ctx
+    -> ((SomeMnemonic, Maybe SomeMnemonic) -> Passphrase "encryption" -> k 'RootK XPrv)
+    -> ApiSharedWalletPostDataFromMnemonics
+    -> Handler ApiSharedWallet
+postSharedWalletFromRootXPrv ctx generateKey body = do
+    let state = mkSharedStateFromRootXPrv (rootXPrv, pwd) accIx g pTemplate dTemplateM
+    void $ liftHandler $ createWalletWorker @_ @s @k ctx wid
+        (\wrk -> W.createWallet  @(WorkerCtx ctx) @s @k wrk wid wName state)
+        (\wrk _ -> W.manageRewardBalance @(WorkerCtx ctx) @s @k (Proxy @n) wrk wid)
+    withWorkerCtx @_ @s @k ctx wid liftE liftE $ \wrk -> liftHandler $
+        W.attachPrivateKeyFromPwd @_ @s @k wrk wid (rootXPrv, pwd)
+    fst <$> getWallet ctx (mkSharedWallet @_ @s @k) (ApiT wid)
+  where
+    seed = getApiMnemonicT (body ^. #mnemonicSentence)
+    secondFactor = getApiMnemonicT <$> (body ^. #mnemonicSecondFactor)
+    pwd = preparePassphrase EncryptWithPBKDF2 $ getApiT (body ^. #passphrase)
+    rootXPrv = generateKey (seed, secondFactor) pwd
+    g = defaultAddressPoolGap
+    accIx = Index $ getDerivationIndex $ getApiT (body ^. #accountIndex)
+    wid = WalletId $ digest $ publicKey rootXPrv
+    pTemplate = body ^. #paymentScriptTemplate
+    dTemplateM = body ^. #delegationScriptTemplate
+    wName = getApiT (body ^. #name)
+
+postSharedWalletFromAccountXPub
+    :: forall ctx s k n.
+        ( s ~ SharedState n k
+        , ctx ~ ApiLayer s k
+        , SoftDerivation k
+        , MkKeyFingerprint k (Proxy n, k 'AddressK XPub)
+        , MkKeyFingerprint k Address
+        , WalletKey k
+        , HasDBFactory s k ctx
+        , HasWorkerRegistry s k ctx
+        , Typeable n
+        )
+    => ctx
+    -> (XPub -> k 'AccountK XPub)
+    -> ApiSharedWalletPostDataFromAccountPubX
+    -> Handler ApiSharedWallet
+postSharedWalletFromAccountXPub ctx liftKey body = do
+    let state = mkSharedStateFromAccountXPub (liftKey accXPub) accIx g pTemplate dTemplateM
+    void $ liftHandler $ createWalletWorker @_ @s @k ctx wid
+        (\wrk -> W.createWallet  @(WorkerCtx ctx) @s @k wrk wid wName state)
+        idleWorker
```

Comment:
Why is this idle in this case :thinking:? It should be possible to also monitor rewards on a shared account.

Reply:
I reckon none of that is covered by the PR anyway, so I would probably change the other one (fromXPrv) to also use a idleWorker and properly address it in a subsequent PR.